src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -3335,7 +3370,19 @@ void InterpreterSelectQuery::executeLimit(QueryPlan & query_plan)
                 throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Negative LIMIT WITH TIES is not supported yet");
         }
 
-        if (!lim_info.is_limit_length_negative && !lim_info.is_limit_offset_negative) [[likely]]
+        if (lim_info.is_limit_length_negative && lim_info.fractional_offset > 0)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Negative Limits can't have a fractional Offset");
```

Comment:
Let's use `NOT_IMPLEMENTED` instead of `BAD_ARGUMENTS`. In the future, we might want to support it as well.
```suggestion
            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Negative LIMIT is not supported with fractional OFFSET");
```